website/payments/signals.py

```diff
@@ -0,0 +1,23 @@
+from django.conf import settings
+from django.db.models.signals import post_save
+
+from members.models import Member
+from payments.models import PaymentUser
+from utils.models.signals import receiver
+
+
+@receiver(post_save, sender=Member)
+def give_new_users_tpay_permissions(instance, created, **kwargs):
+    """If a new user is created, give this user Thalia Pay permissions."""
+    if created and settings.THALIA_PAY_FOR_NEW_MEMBERS:
+        allow_tpay = PaymentUser.allow_tpay
+        allow_tpay(instance)
+
+        # Note that because Django does not support accessing the instance
+        # in the PaymentUser proxy model in this signal, we rely on the fact
```

Comment:
I tried connecting the signal to the ProxyModel but that doesn’t work. It’s still an ongoing django discussion how and whether proxy models signals should be triggered (if you ask me, they should but I see the underlying problem they have). 

I think explaining the code doesn’t hurt and it might prevent future concrexiteers a lot of searching why things behave the way they do